parser/parser.go

```diff
@@ -485,7 +485,13 @@ func (p *Parser) parseBlockStatement() *ast.Statements {
 func (p *Parser) parseFunctionLiteral() ast.Node {
 	lit := &ast.FunctionLiteral{}
 	lit.Token = p.curToken
-
+	// Optional name/identifier
+	if p.peekTokenIs(token.IDENT) {
+		p.nextToken()
+		name := &ast.Identifier{}
+		name.Token = p.curToken
+		lit.Name = name
```

Comment:
It's a bit difficult to read

Maybe this

```suggestion
		lit.Name = &ast.Identifier{Token: p.curToken}
```

Reply:
But the rest of the code might be aligned with current code.

Reply:
it's like that because the Tokens are in Base... so you'd need to do the super annoying {Base: {Token: ...

Reply:
Oh indeed